tools/importer-msgraph-metadata/internal/pipeline/task_translate_service.go

```diff
@@ -0,0 +1,181 @@
+				ResourceIDs: make(map[string]sdkModels.ResourceID),
+			}
+		}
+
+		serviceModels := make(parser.Models)
+
+		// Populate everything else
+		for _, operation := range resource.Operations {
+			var resourceIdName *string
+
+			if operation.ResourceId != nil {
+				resourceIdName = &operation.ResourceId.Name
+
+				sdkResourceId, err := operation.ResourceId.DataApiSdkResourceId()
+				if err != nil {
+					return nil, err
+				}
+
+				sdkServices[resource.Service].APIVersions[resource.Version].Resources[resource.Category].ResourceIDs[operation.ResourceId.Name] = *sdkResourceId
+			}
+
+			var requestObject *sdkModels.SDKObjectDefinition
+			requestObjectIsCommonType := true
+
+			if operation.RequestModel != nil {
+				if !models.Found(*operation.RequestModel) {
+					return nil, fmt.Errorf("request model %q was not found for operation: %s", *operation.RequestModel, operation.Name)
+				}
+
+				if model := models[*operation.RequestModel]; !model.IsValid() {
+					return nil, fmt.Errorf("request model %q was invalid for operation: %s", *operation.RequestModel, operation.Name)
+				} else if !model.Common {
+					requestObjectIsCommonType = false
+
+					if err := serviceModels.MergeDependants(models, *operation.RequestModel); err != nil {
+						return nil, err
+					}
+				}
+
+				requestObject = &sdkModels.SDKObjectDefinition{
+					ReferenceName:             operation.RequestModel,
+					ReferenceNameIsCommonType: &requestObjectIsCommonType,
+					Type:                      sdkModels.ReferenceSDKObjectDefinitionType,
+				}
+			} else if operation.RequestType != nil {
+				requestObject = &sdkModels.SDKObjectDefinition{
+					Type: operation.RequestType.DataApiSdkObjectDefinitionType(),
+				}
+			}
+
+			var responseObject *sdkModels.SDKObjectDefinition
+			responseObjectIsCommonType := true
+
+			for _, response := range operation.Responses {
+				if response.Type != nil && *response.Type == parser.DataTypeModel && response.ModelName != nil {
+					if !models.Found(*response.ModelName) {
+						return nil, fmt.Errorf("response model %q was not found for operation: %s", *response.ModelName, operation.Name)
+					}
+
+					if model := models[*response.ModelName]; !model.IsValid() {
+						return nil, fmt.Errorf("response model %q was invalid for operation: %s", *response.ModelName, operation.Name)
+					} else if !model.Common {
+						responseObjectIsCommonType = false
+
+						if err := serviceModels.MergeDependants(models, *response.ModelName); err != nil {
+							return nil, err
+						}
+					}
+
+					responseObject = &sdkModels.SDKObjectDefinition{
+						ReferenceName:             response.ModelName,
+						ReferenceNameIsCommonType: &responseObjectIsCommonType,
+						Type:                      sdkModels.ReferenceSDKObjectDefinitionType,
+					}
+				} else if response.Type != nil {
+					responseObject = &sdkModels.SDKObjectDefinition{
+						Type: response.Type.DataApiSdkObjectDefinitionType(),
```

Comment:
do we need to set `ReferenceNameIsCommonType: &responseObjectIsCommonType` here?

Reply:
As with the request object, `response.Type` always references a builtin type